src/node_options.cc

```diff
@@ -168,6 +168,19 @@ void EnvironmentOptions::CheckOptions(std::vector<std::string>* errors) {
     }
   }
 
+  if (!heap_prof) {
+    if (!heap_prof_name.empty()) {
+      errors->push_back("--heap-prof-name must be used with --heap-prof");
```

Comment:
Just for my own understanding: Are we expecting to use these in the future for runtime profiles as well? I assume at that point we'd remove the restriction..?

Reply:
By runtime profile do you mean an API to do this on-demand? In that case I guess the restriction would be natural with the API design. Take the snippet in https://github.com/nodejs/node/issues/27421#issuecomment-486980997 for an example: I'd assume the profile name would be part of the options passed to an exporter, and it would be quite different from the current model where all options are passed together in the command line.

Reply:
Thanks for the additional context, that makes sense! 👍

I thought about some sort of simple external signal to trigger this but it would realistically be a full API that could easily pass these values.

Reply:
@jkrems Currently i didn't add the possibility to have custom name for a profile but i added it as an issue [there](https://github.com/vmarchaud/openprofiling-node/issues/1)